src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/PackageArgumentFactoryUtility.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.CommandLine.XPlat.Commands.Package;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    internal static class PackageArgumentFactoryUtility
```

Comment:
The factory isn't needed for generics, instead you can create a static method on the class and have it create typed instances

https://dotnetfiddle.net/jFHQhx

```c#
public static PackageArgument<TVersion> Create(string id, TVersion version, IEqualityComparer<TVersion?> versionComparer)
{
    return new PackageArgument<TVersion>(versionComparer)
	{
	    Id = id,
        Version = version,
    };
}
```